packages/flame_riverpod/README.md

```diff
@@ -0,0 +1,76 @@
+# flame_riverpod
+
+Riverpod is a reactive caching and data-binding framework for Dart & Flutter.
+
+In `flutter_riverpod`, widgets can be configured to rebuild when the state
+of a provider changes.
+
+When using Flame, we are interacting with components, which are *not* Widgets.
+
+`flame_riverpod` provides the `ComponentRef` and `HasComponentRef` to
+facilitate managing state from Providers in your Flame Game.
+
+
+## Usage
+
+Your Widget that extends `FlameGame` should use the HasComponentRef mixin,
+and accept a WidgetRef inside its constructor, and should be wrapped in a
```

Comment:
Is the requirement that the `MyGame` class derived from `FlameGame` must have a single parameter in the constructor which is a `WidgetRef`? This seems overly restrictive, what if the user needs to add some other parameters into the constructor?

Reply:
The expectation is that a WidgetRef is provided within the constructor. It need not be the only parameter. IMO a user consuming the package will see this from the usage, but open to suggestions on rewording this description.

Reply:
Is `WidgetRef` even necessary? Why can't `ComponentRef` talk to the `riverpod` backend directly?

Reply:
> Why can't ComponentRef talk to the riverpod backend directly?

Do you mean by using a `ProviderContainer`? Since most people use `ProviderScope` in Flutter I don't think that is a good idea.

Reply:
Ah, or maybe you are correct, maybe that is the second point that @rrousselGit is mentioning [here](https://github.com/flame-engine/flame/pull/2367#discussion_r1121528187).
I don't know enough about the internals of Riverpod and how the `WidgetRef` works, but I assumed that we'd need some connection to the `WidgetRef` to be able to listen to the same providers as are used in the Flutter tree.

Reply:
Connecting to the ProviderContainer directly (though `ProviderScope.containerOf(context)`) and making your own ref from scratch is the most logical solution

Reply:
That sounds like a great solution, then there would be less boilerplate för the user too right?